# Het afgeleid getal met Python

De bibliotheek `sympy` rekent met formules in plaats van met getallen. Ze kan een limiet uitrekenen, dus kan ze het afgeleid getal rechtstreeks uit de definitie bepalen. Dat is een goede controle van je berekening, maar geen vervanging ervan.

- Schrijf het voorschrift als een functie, zoals hierboven. Dan mag je in Python net als op papier $f(a)$ schrijven.
- Zeg met `symbols` dat `x` een veranderlijke is. Zo mag ook $f(x)$, met een formule als resultaat in plaats van een getal.

In [ ]:
from sympy import symbols, limit

def f(x):
    return x**2 / 4

x = symbols('x')
a = 3

print(f(a))
print((f(x) - f(a)) / (x - a))

2.25
(x**2/4 - 2.25)/(x - 3)


Het differentiequotiënt $\dfrac {f(x)-f(a)}{x-a}$ schrijf je dus letterlijk zo op.

- `limit(uitdrukking, x, a)` berekent $\lim _{x\to a}$ van die uitdrukking. Met `float` eromheen krijg je een kommagetal.

In [ ]:
m = float(limit((f(x) - f(a)) / (x - a), x, a))
print(m)

1.5


Python vindt dus $f'(3) = 1.5$, hetzelfde als hierboven. Verander je `a`, dan krijg je het afgeleid getal in een ander punt.